### ISSUE TITLE
Typing of Exceptions classes seems incorrect

### ISSUE THREAD

**Issue by MartinAltmayer** [vemel/mypy_boto3_builder#10]
**Describe the bug**
I believe the type annotations of the various "Exceptions" classes are incorrect: E.g.
```
class Exceptions:
    BucketAlreadyExists: Boto3ClientError
    BucketAlreadyOwnedByYou: Boto3ClientError
```
tells Mypy that  `exceptions.BucketAlreadyExists` is an instance of `BotoClientError`, when in reality it is a subclass. Wouldn't 
```
class Exceptions:
    BucketAlreadyExists: Type[Boto3ClientError]
    BucketAlreadyOwnedByYou: Type[Boto3ClientError]
```
be correct?

For some reason, mypy only notices this error in the presence of stub files for botocore.exceptions (see the following steps).

**To Reproduce**
1. Install boto3-stubs[s3]

2. Write a script `test.py`:
```
import boto3
from mypy_boto3 import s3

client: s3.Client = boto3.client('s3')
try:
    client.list_objects(Bucket='my-test-bucket-7f327deb-2584-4754-99ba-c01f52225fa7')
except client.exceptions.NoSuchBucket:
    print("Bucket not found")
```
3. Create a directory stubs/botocore with an empty `__init__.pyi` and a file `exceptions.pyi` with the following content:
```
from typing import Dict, Any

class ClientError(Exception):
    def __init__(self, error_response: Dict[str, Any], operation_name: str): ...
```
4. Configure Mypy to use the stubs by putting these lines in setup.cfg:
```
[mypy]
mypy_path=stubs
```
5. Running `mypy test.py` now gives this error:
```
test.py:7: error: Exception type must be derived from BaseException
Found 1 error in 1 file (checked 1 source file)
```

**Additional context**
mypy 0.761
boto3-stubs           1.12.27.0

**Comment by vemel** [comment]
Thank you for the report. It should be easy to fix.

**Comment by vemel** [comment]
Should be fixed in the latest `boto3-stubs==1.12.28.1`. I also recommend to update `boto3` to `1.12.28` to keep type annotations consistent.

Please let me know if the issue still exists or if you find something related.

**Comment by MartinAltmayer** [comment]
Thanks for fixing this so quickly!
